### PR TITLE
feat: Unblock user

### PR DESCRIPTION
**Pull request by borichellow**
[Task-2084](https://wearezeta.atlassian.net/browse/AR-2084)

# What's new in this PR?

Added availability to un-block blocked user

### Attachments (Optional)

https://user-images.githubusercontent.com/6539347/185391860-a12929d4-950a-4c0b-80ca-70f354af37ad.mp4

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-856/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-856/2/) **failed**.

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-856/3/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-856/4/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4186087)(4186087)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4186087)(4186087)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-856/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4186716)(4186716)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4186716)(4186716)-dev-release.apk)

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -540,4 +541,9 @@
     <string name="block_user_dialog_body">%1$s won’t be able to contact you or add you to group conversations. You will lose access to any 1:1 conversations with them. The other user will not be notified about this.</string>
     <string name="block_user_dialog_confirm_button">Block</string>
 
+    <!--unblock user dialog -->
+    <string name="unblock_user_dialog_title">Unblock user?</string>
+    <string name="unblock_user_dialog_body">%1$s will be able to contact you or add you to group conversations. You will be able to access a 1:1 conversation with them.</string>
+    <string name="unblock_user_dialog_confirm_button">Unlock</string>
```
Comment (saleniuk) [inline app/src/main/res/values/strings.xml:547]: typo - should be "Unblock"?

File: app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreen.kt [hunk 2]
```diff
@@ -413,9 +368,8 @@ private fun ContentFooter(
                         onOpenConversation,
                         onCancelConnectionRequest,
                         acceptConnectionRequest,
-                        ignoreConnectionRequest,
-                        onUnblockUser
-                    )
+                        ignoreConnectionRequest
+                    ) { onUnblockUser(state.userName) }
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreen.kt:372]: Why username and not user Id?

File: app/src/main/kotlin/com/wire/android/ui/common/dialogs/UnblockUserDialogs.kt [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            dismissButtonProperties = WireDialogButtonProperties(
+                onClick = dismiss,
+                text = stringResource(id = android.R.string.cancel),
+                type = WireDialogButtonType.Secondary,
+            )
+        )
+    }
+}
+
+data class UnblockUserDialogState(val userName: String, val userId: UserId)
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/ui/common/dialogs/UnblockUserDialogs.kt:55]:
```suggestion
data class UnblockUserDialogState(val username: String, val userId: UserId)
```

File: app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreen.kt (hunk 2 above)
Comment (borichellow) [reply app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreen.kt:372]: `userId` is needed for unblocking the user - yes, but it's a `val` in ViewModel, not need duplicate it here
And `username` is needed to display a correct error, if it occurs

**Review by gongracr — APPROVED** [review]
Assuming you'll address changes requested from Michal I give you my early blessings ✝️

File: app/src/main/kotlin/com/wire/android/ui/common/dialogs/UnblockUserDialogs.kt [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+import com.wire.android.util.ui.stringWithStyledArgs
+import com.wire.kalium.logic.data.user.UserId
+
+@Composable
+fun UnblockUserDialogContent(
+    dialogState: PreservedState<UnblockUserDialogState>?,
+    dismiss: () -> Unit = {},
+    onUnblock: (UserId) -> Unit = { }
+) {
+    VisibilityState(dialogState) { preservedState ->
```
[the remark this answers is not in the document]
Comment (borichellow) [reply app/src/main/kotlin/com/wire/android/ui/common/dialogs/UnblockUserDialogs.kt:25]: cause it makes less code in Screen, I think :) Why not?

**Comment by AndroidBob** [comment]
Build [6](https://10.10.124.134/job/wire-android-reloaded/job/PR-856/6/) **failed**.

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-856/7/) **failed**.

**Comment by AndroidBob** [comment]
Build [8](https://10.10.124.134/job/wire-android-reloaded/job/PR-856/8/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4219758)(4219758)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4219758)(4219758)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [9](https://10.10.124.134/job/wire-android-reloaded/job/PR-856/9/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4219992)(4219992)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4219992)(4219992)-dev-release.apk)

**Comment by AndroidBob** [comment]
Build [10](https://10.10.124.134/job/wire-android-reloaded/job/PR-856/10/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4220208)(4220208)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4220208)(4220208)-dev-release.apk)

**Pull request merged**